mmv1/products/edgenetwork/Router.yaml

```diff
@@ -0,0 +1,227 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'Router'
+description: |
+  A Distributed Cloud Edge router, which encapsulates northbound layer-3 parameters for a zone.
+references:
+  guides:
+    'Create and manage routers': 'https://cloud.google.com/distributed-cloud/edge/latest/docs/routers#api'
+  api: 'https://cloud.google.com/distributed-cloud/edge/latest/docs/reference/network/rest/v1/projects.locations.zones.routers'
+docs:
+base_url: 'projects/{{project}}/locations/{{location}}/zones/{{zone}}/routers'
+self_link: 'projects/{{project}}/locations/{{location}}/zones/{{zone}}/routers/{{router_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/zones/{{zone}}/routers?routerId={{router_id}}'
+update_verb: 'PATCH'
+update_mask: true
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/zones/{{zone}}/routers/{{router_id}}'
+  - '{{name}}'
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 30
+autogen_async: true
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    path: 'name'
+    wait_ms: 1000
+  result:
+    path: 'response'
+    resource_inside_response: false
+  error:
+    path: 'error'
+    message: 'message'
+examples:
+  - name: 'edgenetwork_router_minimal'
+    primary_resource_id: 'example_router'
+    vars:
+      edgenetwork_router_id: 'example-router'
+      edgenetwork_network_id: 'example-network'
+    test_env_vars:
+      location: 'REGION'
+      zone: 'ZONE'
+    exclude_test: true
+  - name: 'edgenetwork_router_with_bgp'
+    primary_resource_id: 'example_router'
+    vars:
+      edgenetwork_router_id: 'example-router'
+      edgenetwork_interconnect_attachment_id: 'example-interconnect-attachment'
+      edgenetwork_network_id: 'example-network'
+      edgenetwork_subnet_id: 'example-subnet'
+    test_env_vars:
+      location: 'REGION'
+      zone: 'ZONE'
+    exclude_test: true
+parameters:
+  - name: 'location'
+    type: String
+    description: |
+      The Google Cloud region to which the target Distributed Cloud Edge zone belongs.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'zone'
+    type: String
+    description: |
+      The name of the target Distributed Cloud Edge zone.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'router_id'
+    type: String
+    description: |
+      A unique ID that identifies this router.
+    url_param_only: true
+    required: true
+    immutable: true
+properties:
+  - name: 'name'
+    type: String
+    description: |
+      The canonical name of this resource, with format
+      `projects/{{project}}/locations/{{location}}/zones/{{zone}}/routers/{{router_id}}`
+    output: true
+  - name: 'labels'
+    type: KeyValueLabels
+    description: |
+      Labels associated with this resource.
+    required: false
```

Comment:
Nit: `required: false` is the default and can be removed in numerous places here

Reply:
Fixed